ftw/solr/document.py

```diff
@@ -0,0 +1,64 @@
+from ftw.solr.interfaces import ISolrDocument
+from zope.component.hooks import getSite
+from zope.globalrequest import getRequest
+from zope.interface import implementer
+
+
+@implementer(ISolrDocument)
+class SolrDocument(object):
+
+    def __init__(self, data, fields=None):
+        self.data = unicode2bytes(data)
+
+    def __getitem__(self, key):
+        return self.data[key]
+
+    def __setitem__(self, key, value):
+        self.data[key] = value
+
+    def __contains__(self, item):
+        return item in self.data
+
+    def get(self, key, default=None):
+        return self.data.get(key, default)
+
+    def __getattr__(self, name):
+        if name in self.data:
+            return self.data[name]
+        else:
+            raise AttributeError
+
+    @property
+    def getId(self):
+        return self.data.get('id')
+
+    def getPath(self):
+        return self.path
+
+    def getURL(self, relative=False):
+        request = getRequest()
+        return request.physicalPathToURL(self.getPath(), relative)
+
+    def getObject(self, REQUEST=None, restricted=True):
+        site = getSite()
+        path = self.getPath()
+        if not path:
+            return None
+        path = path.split('/')
+        if restricted:
+            parent = site.unrestrictedTraverse(path[:-1], None)
+            if parent is None:
+                return None
+            return parent.restrictedTraverse(path[-1], None)
+        return site.unrestrictedTraverse(path, None)
```

Comment:
When something is outdated it could happen that we get an object here, but it's path is not `path`.
This can happen when there is no object at `path`, but there is an uncle with the same id - acquisition will return another object then.
Should we handle this case by checking that the traversed object actually has the path we expect it to have?

I ran into this problem with the catalog now and then..

Reply:
Good point. I've added a check to make sure we only return an object if it's path matches.